qos/jsonrpc/request_test.go

```diff
@@ -0,0 +1,36 @@
+package jsonrpc
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TODO_MVP(@adshmh): add a test case for batch JSONRPC requests
+func TestUnmarshalParams(t *testing.T) {
+	testCases := []struct {
+		name       string
+		rawPayload []byte
+	}{
+		{
+			name:       "params as array of single object",
```

Comment:
Can you add a few tests:

1. Params as an empty object
2. Params as an empty list
3. Params as just an object (not surrounded by an array)